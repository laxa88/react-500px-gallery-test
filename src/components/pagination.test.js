import React from 'react';
import 'jest';
import 'enzyme';

import Pagination from './pagination';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// import configureStore from 'redux-mock-store';
// import * as C from '../constants';

configure({adapter: new Adapter()});

// const mockStore = configureStore();

describe('Pagination', () => {
  test('Can render', () => {
    const wrapper = shallow(
      <Pagination
        onClick={jest.fn()}
        page={1}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('Can render range of pages', () => {
    const wrapper = shallow(
      <Pagination
        onClick={jest.fn()}
        page={10}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('Calls onClick', () => {
    const callback = jest.fn();

    const wrapper = shallow(
      <Pagination
        onClick={callback}
        page={1}
      />
    );

    wrapper.instance().handleClick({target: {innerHTML: 10}});

    expect(callback).toHaveBeenCalledTimes(1);
  });
});
