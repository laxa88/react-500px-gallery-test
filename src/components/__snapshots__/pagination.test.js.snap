// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Can render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination
    onClick={[MockFunction]}
    page={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          &lt;
        </div>,
        Array [
          <div>
            1
          </div>,
          <div>
            <a
              href="#"
              onClick={[Function]}
            >
              2
            </a>
          </div>,
          <div>
            <a
              href="#"
              onClick={[Function]}
            >
              3
            </a>
          </div>,
          <div>
            <a
              href="#"
              onClick={[Function]}
            >
              4
            </a>
          </div>,
          <div>
            <a
              href="#"
              onClick={[Function]}
            >
              5
            </a>
          </div>,
          <div>
            <a
              href="#"
              onClick={[Function]}
            >
              6
            </a>
          </div>,
        ],
        <div>
          &gt;
        </div>,
      ],
      "className": "pagination",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "<",
        },
        "ref": null,
        "rendered": "<",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": 1,
        },
        "ref": null,
        "rendered": 1,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#"
            onClick={[Function]}
          >
            2
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 2,
            "href": "#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 2,
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#"
            onClick={[Function]}
          >
            3
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 3,
            "href": "#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 3,
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#"
            onClick={[Function]}
          >
            4
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 4,
            "href": "#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 4,
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#"
            onClick={[Function]}
          >
            5
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 5,
            "href": "#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 5,
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#"
            onClick={[Function]}
          >
            6
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 6,
            "href": "#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 6,
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": ">",
        },
        "ref": null,
        "rendered": ">",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            &lt;
          </div>,
          Array [
            <div>
              1
            </div>,
            <div>
              <a
                href="#"
                onClick={[Function]}
              >
                2
              </a>
            </div>,
            <div>
              <a
                href="#"
                onClick={[Function]}
              >
                3
              </a>
            </div>,
            <div>
              <a
                href="#"
                onClick={[Function]}
              >
                4
              </a>
            </div>,
            <div>
              <a
                href="#"
                onClick={[Function]}
              >
                5
              </a>
            </div>,
            <div>
              <a
                href="#"
                onClick={[Function]}
              >
                6
              </a>
            </div>,
          ],
          <div>
            &gt;
          </div>,
        ],
        "className": "pagination",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "<",
          },
          "ref": null,
          "rendered": "<",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": 1,
          },
          "ref": null,
          "rendered": 1,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
              onClick={[Function]}
            >
              2
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 2,
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 2,
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
              onClick={[Function]}
            >
              3
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 3,
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 3,
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
              onClick={[Function]}
            >
              4
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 4,
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 4,
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
              onClick={[Function]}
            >
              5
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 5,
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 5,
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
              onClick={[Function]}
            >
              6
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 6,
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 6,
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": ">",
          },
          "ref": null,
          "rendered": ">",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Pagination Can render range of pages 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination
    onClick={[MockFunction]}
    page={10}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          &lt;
        </div>,
        Array [
          <div>
            <a
              href="#"
              onClick={[Function]}
            >
              5
            </a>
          </div>,
          <div>
            <a
              href="#"
              onClick={[Function]}
            >
              6
            </a>
          </div>,
          <div>
            <a
              href="#"
              onClick={[Function]}
            >
              7
            </a>
          </div>,
          <div>
            <a
              href="#"
              onClick={[Function]}
            >
              8
            </a>
          </div>,
          <div>
            <a
              href="#"
              onClick={[Function]}
            >
              9
            </a>
          </div>,
          <div>
            10
          </div>,
          <div>
            <a
              href="#"
              onClick={[Function]}
            >
              11
            </a>
          </div>,
          <div>
            <a
              href="#"
              onClick={[Function]}
            >
              12
            </a>
          </div>,
          <div>
            <a
              href="#"
              onClick={[Function]}
            >
              13
            </a>
          </div>,
          <div>
            <a
              href="#"
              onClick={[Function]}
            >
              14
            </a>
          </div>,
          <div>
            <a
              href="#"
              onClick={[Function]}
            >
              15
            </a>
          </div>,
        ],
        <div>
          &gt;
        </div>,
      ],
      "className": "pagination",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "<",
        },
        "ref": null,
        "rendered": "<",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#"
            onClick={[Function]}
          >
            5
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 5,
            "href": "#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 5,
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#"
            onClick={[Function]}
          >
            6
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 6,
            "href": "#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 6,
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "7",
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#"
            onClick={[Function]}
          >
            7
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 7,
            "href": "#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 7,
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "8",
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#"
            onClick={[Function]}
          >
            8
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 8,
            "href": "#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 8,
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "9",
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#"
            onClick={[Function]}
          >
            9
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 9,
            "href": "#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 9,
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "10",
        "nodeType": "host",
        "props": Object {
          "children": 10,
        },
        "ref": null,
        "rendered": 10,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "11",
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#"
            onClick={[Function]}
          >
            11
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 11,
            "href": "#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 11,
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "12",
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#"
            onClick={[Function]}
          >
            12
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 12,
            "href": "#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 12,
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "13",
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#"
            onClick={[Function]}
          >
            13
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 13,
            "href": "#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 13,
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "14",
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#"
            onClick={[Function]}
          >
            14
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 14,
            "href": "#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 14,
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "15",
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#"
            onClick={[Function]}
          >
            15
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 15,
            "href": "#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 15,
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": ">",
        },
        "ref": null,
        "rendered": ">",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            &lt;
          </div>,
          Array [
            <div>
              <a
                href="#"
                onClick={[Function]}
              >
                5
              </a>
            </div>,
            <div>
              <a
                href="#"
                onClick={[Function]}
              >
                6
              </a>
            </div>,
            <div>
              <a
                href="#"
                onClick={[Function]}
              >
                7
              </a>
            </div>,
            <div>
              <a
                href="#"
                onClick={[Function]}
              >
                8
              </a>
            </div>,
            <div>
              <a
                href="#"
                onClick={[Function]}
              >
                9
              </a>
            </div>,
            <div>
              10
            </div>,
            <div>
              <a
                href="#"
                onClick={[Function]}
              >
                11
              </a>
            </div>,
            <div>
              <a
                href="#"
                onClick={[Function]}
              >
                12
              </a>
            </div>,
            <div>
              <a
                href="#"
                onClick={[Function]}
              >
                13
              </a>
            </div>,
            <div>
              <a
                href="#"
                onClick={[Function]}
              >
                14
              </a>
            </div>,
            <div>
              <a
                href="#"
                onClick={[Function]}
              >
                15
              </a>
            </div>,
          ],
          <div>
            &gt;
          </div>,
        ],
        "className": "pagination",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "<",
          },
          "ref": null,
          "rendered": "<",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
              onClick={[Function]}
            >
              5
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 5,
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 5,
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
              onClick={[Function]}
            >
              6
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 6,
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 6,
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
              onClick={[Function]}
            >
              7
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 7,
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 7,
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "8",
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
              onClick={[Function]}
            >
              8
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 8,
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 8,
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "9",
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
              onClick={[Function]}
            >
              9
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 9,
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 9,
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "10",
          "nodeType": "host",
          "props": Object {
            "children": 10,
          },
          "ref": null,
          "rendered": 10,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "11",
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
              onClick={[Function]}
            >
              11
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 11,
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 11,
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "12",
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
              onClick={[Function]}
            >
              12
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 12,
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 12,
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "13",
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
              onClick={[Function]}
            >
              13
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 13,
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 13,
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "14",
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
              onClick={[Function]}
            >
              14
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 14,
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 14,
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "15",
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
              onClick={[Function]}
            >
              15
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 15,
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 15,
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": ">",
          },
          "ref": null,
          "rendered": ">",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
